apps/explorer/src/ui/ExpandableList.tsx

```diff
@@ -26,27 +31,48 @@ export function ExpandableList({
     const handleShowAllClick = () =>
         setShowAll((prevShowAll: boolean) => !prevShowAll);
 
+    let showAllText = '';
+    if (showAll) {
+        showAllText = 'Show Less';
+    } else {
+        showAllText = itemsLabel
+            ? `Show All ${items.length} ${itemsLabel}`
+            : 'Show All';
+    }
+
+    const renderItemsDisplayed = itemsDisplayed.map((item, index) => (
+        <div key={index}>{item}</div>
+    ));
+
     return (
         <>
-            {itemsDisplayed.map((item, index) => (
-                <div key={index}>{item}</div>
-            ))}
+            {contentWrapperClassName && items.length > defaultItemsToShow ? (
+                <div
+                    className={clsx(
+                        'flex flex-col overflow-y-auto',
+                        contentWrapperClassName
```

Comment:
@Jordan-Mysten not 100% sure how to handle this case here. The `ExpandableList` is accepting an array and we will need a wrapper of the internal nodes to set the max height correctly.